src/main/java/WordFinder.java

```diff
@@ -35,8 +42,11 @@ private String findWord(String dictionaryWord, String wordJumble) {
         return joinedMatchingLetters.length() == dictionaryWord.length() ? joinedMatchingLetters : null;
     }
 
+    //passing matching letters as a param?
     private void findMatchingLetters(List<String> splitDictionaryWord, List<String> splitWordJumble, List<String> matchingLetters) {
```

Comment:
Surely matching letters should be returned rather than passed as a param.  As a general rule I try to avoid modifying values passed as parameters as it's not very obvious that they're changing.